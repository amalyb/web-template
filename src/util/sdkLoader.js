import * as importedSdk from 'sharetribe-flex-sdk';

let exportSdk;

const isServer = () => typeof window === 'undefined';

if (isServer()) {
  // Use eval to skip webpack from bundling SDK in Node
  // eslint-disable-next-line no-eval
  exportSdk = eval('require')('sharetribe-flex-sdk');
} else {
  exportSdk = importedSdk;
}

const { createInstance, types, transit, util } = exportSdk;

// create image variant from variant name, desired width and aspectRatio
const createImageVariantConfig = (name, width, aspectRatio) => {
  //const aspectRatio = aspectHeight / aspectWidth;
  return {
    [`imageVariant.${name}`]: util.objectQueryString({
      w: width,
      h: aspectRatio * width,
      fit: 'crop',
    }),
  };
};

export { createInstance, types, transit, util, createImageVariantConfig };
