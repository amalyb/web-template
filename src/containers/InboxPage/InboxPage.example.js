import React from 'react';

import { injectIntl } from '../../util/reactIntl';
import { createUser, createTransaction } from '../../util/test-data';
import { getProcess } from '../../util/transaction';
import { LINE_ITEM_UNITS } from '../../util/types';

import { InboxItem, txState } from './InboxPage';

const noop = () => null;
const transitions = getProcess('flex-product-default-process')?.transitions;

const tx = lastTransition =>
  createTransaction({
    id: 'order-1',
    lastTransition,
    lastTransitionedAt: new Date(Date.UTC(2017, 0, 15)),
    customer: createUser('John Customer', {
      banned: false,
      deleted: false,
      profile: {
        displayName: 'John C',
        abbreviatedName: 'JC',
      },
    }),
    provider: createUser('Jane Provider', {
      banned: false,
      deleted: false,
      profile: {
        displayName: 'Jane P',
        abbreviatedName: 'JP',
      },
    }),
  });

const TranslatedInboxItem = props => {
  const { intl, tx, type, ...rest } = props;
  const stateData = txState(intl, tx, type);

  return <InboxItem intl={intl} tx={tx} stateData={stateData} type={type} {...rest} />;
};

export const EnquiryPayment = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.ENQUIRE),
  },
  group: 'page:InboxPage',
};

///////////
// Order //
///////////

export const PendingPayment_Order = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.REQUEST_PAYMENT),
  },
  group: 'page:InboxPage',
};

export const Purchased_Order = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.CONFIRM_PAYMENT),
  },
  group: 'page:InboxPage',
};

export const Canceled_Order = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.CANCEL),
  },
  group: 'page:InboxPage',
};

export const Delivered_Order = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.MARK_DELIVERED),
  },
  group: 'page:InboxPage',
};

export const Disputed_Order = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.DISPUTE),
  },
  group: 'page:InboxPage',
};

export const Received_Order = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.MARK_RECEIVED),
  },
  group: 'page:InboxPage',
};

export const ReviewedByCustomer_Order = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.REVIEW_1_BY_CUSTOMER),
  },
  group: 'page:InboxPage',
};

export const ReviewedByProvider_Order = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.REVIEW_1_BY_PROVIDER),
  },
  group: 'page:InboxPage',
};

export const Reviewed_Order = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'order',
    tx: tx(transitions.EXPIRE_REVIEW_PERIOD),
  },
  group: 'page:InboxPage',
};

//////////
// Sale //
//////////

export const PendingPayment_Sale = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'sale',
    tx: tx(transitions.REQUEST_PAYMENT),
  },
  group: 'page:InboxPage',
};

export const Purchased_Sale = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'sale',
    tx: tx(transitions.CONFIRM_PAYMENT),
  },
  group: 'page:InboxPage',
};

export const Canceled_Sale = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'sale',
    tx: tx(transitions.CANCEL),
  },
  group: 'page:InboxPage',
};

export const Delivered_Sale = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'sale',
    tx: tx(transitions.MARK_DELIVERED),
  },
  group: 'page:InboxPage',
};

export const Disputed_Sale = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'sale',
    tx: tx(transitions.DISPUTE),
  },
  group: 'page:InboxPage',
};

export const Received_Sale = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'sale',
    tx: tx(transitions.MARK_RECEIVED),
  },
  group: 'page:InboxPage',
};

export const ReviewedByCustomer_Sale = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'sale',
    tx: tx(transitions.REVIEW_1_BY_CUSTOMER),
  },
  group: 'page:InboxPage',
};

export const ReviewedByProvider_Sale = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'sale',
    tx: tx(transitions.REVIEW_1_BY_PROVIDER),
  },
  group: 'page:InboxPage',
};

export const Reviewed_Sale = {
  component: injectIntl(TranslatedInboxItem),
  props: {
    unitType: LINE_ITEM_UNITS,
    type: 'sale',
    tx: tx(transitions.EXPIRE_REVIEW_PERIOD),
  },
  group: 'page:InboxPage',
};
