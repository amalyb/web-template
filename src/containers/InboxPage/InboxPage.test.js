import React from 'react';
import Decimal from 'decimal.js';

import { renderShallow, renderDeep } from '../../util/test-helpers';
import {
  fakeIntl,
  createCurrentUser,
  createUser,
  createListing,
  createTransaction,
} from '../../util/test-data';
import { types as sdkTypes } from '../../util/sdkLoader';
import {
  TX_TRANSITION_ACTOR_CUSTOMER,
  TX_TRANSITION_ACTOR_PROVIDER,
  getProcess,
} from '../../util/transaction';
import { LINE_ITEM_ITEM, LINE_ITEM_PROVIDER_COMMISSION } from '../../util/types';

import { getStateData } from './InboxPage.stateData';
import { InboxPageComponent, InboxItem } from './InboxPage';

const { Money } = sdkTypes;
const noop = () => null;
const transitions = getProcess('flex-product-default-process')?.transitions;

describe('InboxPage', () => {
  it('matches snapshot', () => {
    const provider = createUser('provider-user-id');
    const customer = createUser('customer-user-id');
    const currentUserProvider = createCurrentUser('provider-user-id');
    const currentUserCustomer = createCurrentUser('customer-user-id');
    const listing = createListing('ItemX', {
      publicData: { transactionProcessAlias: 'flex-product-default-process', unitType: 'item' },
    });

    const lineItems = [
      {
        code: LINE_ITEM_ITEM,
        includeFor: ['customer', 'provider'],
        quantity: new Decimal(1),
        unitPrice: new Money(1000, 'USD'),
        lineTotal: new Money(1000, 'USD'),
        reversal: false,
      },
      {
        code: LINE_ITEM_PROVIDER_COMMISSION,
        includeFor: ['provider'],
        unitPrice: new Money(100 * -1, 'USD'),
        lineTotal: new Money(100 * -1, 'USD'),
        reversal: false,
      },
    ];

    const ordersProps = {
      location: { search: '' },
      history: {
        push: () => console.log('HistoryPush called'),
      },
      params: {
        tab: 'orders',
      },
      authInProgress: false,
      currentUser: currentUserProvider,
      currentUserHasListings: false,
      isAuthenticated: false,
      fetchInProgress: false,
      onLogout: noop,
      onManageDisableScrolling: noop,
      transactions: [
        createTransaction({
          id: 'order-1',
          lastTransition: transitions.CONFIRM_PAYMENT,
          customer,
          provider,
          listing,
          lastTransitionedAt: new Date(Date.UTC(2017, 0, 15)),
          lineItems,
        }),
        createTransaction({
          id: 'order-2',
          lastTransition: transitions.CONFIRM_PAYMENT,
          customer,
          provider,
          listing,
          lastTransitionedAt: new Date(Date.UTC(2016, 0, 15)),
          lineItems,
        }),
      ],
      intl: fakeIntl,
      scrollingDisabled: false,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
    };

    const ordersTree = renderShallow(<InboxPageComponent {...ordersProps} />);
    expect(ordersTree).toMatchSnapshot();

    const stateDataOrder = getStateData({
      transaction: ordersProps.transactions[0],
      transactionRole: TX_TRANSITION_ACTOR_CUSTOMER,
    });

    // Deeply render one InboxItem
    const orderItem = renderDeep(
      <InboxItem
        tx={ordersProps.transactions[0]}
        transactionRole={TX_TRANSITION_ACTOR_CUSTOMER}
        intl={fakeIntl}
        stateData={stateDataOrder}
      />
    );
    expect(orderItem).toMatchSnapshot();

    const salesProps = {
      location: { search: '' },
      history: {
        push: () => console.log('HistoryPush called'),
      },
      params: {
        tab: 'sales',
      },
      authInProgress: false,
      currentUser: currentUserCustomer,
      currentUserHasListings: false,
      isAuthenticated: false,
      fetchInProgress: false,
      onLogout: noop,
      onManageDisableScrolling: noop,
      transactions: [
        createTransaction({
          id: 'sale-1',
          lastTransition: transitions.CONFIRM_PAYMENT,
          customer,
          provider,
          listing,
          lastTransitionedAt: new Date(Date.UTC(2017, 0, 15)),
          lineItems,
        }),
        createTransaction({
          id: 'sale-2',
          lastTransition: transitions.CONFIRM_PAYMENT,
          customer,
          provider,
          listing,
          lastTransitionedAt: new Date(Date.UTC(2016, 0, 15)),
          lineItems,
        }),
      ],
      intl: fakeIntl,
      scrollingDisabled: false,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
    };

    const salesTree = renderShallow(<InboxPageComponent {...salesProps} />);
    expect(salesTree).toMatchSnapshot();

    const stateDataSale = getStateData({
      transaction: salesProps.transactions[0],
      transactionRole: TX_TRANSITION_ACTOR_PROVIDER,
    });

    // Deeply render one InboxItem
    const saleItem = renderDeep(
      <InboxItem
        type="sale"
        tx={salesProps.transactions[0]}
        transactionRole={TX_TRANSITION_ACTOR_PROVIDER}
        intl={fakeIntl}
        stateData={stateDataSale}
      />
    );
    expect(saleItem).toMatchSnapshot();
  });
});
