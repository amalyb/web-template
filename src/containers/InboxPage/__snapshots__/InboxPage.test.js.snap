// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InboxPage InboxItem matches snapshot of order 1`] = `
<div
  className="item"
>
  <div
    className="itemAvatar"
  >
    <a
      className="root"
      href="/u/provider-user-id"
      onClick={[Function]}
      onMouseOver={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
      title="provider-user-id display name"
    >
      <span
        className="initials"
      >
        TT
      </span>
    </a>
  </div>
  <a
    className="itemLink"
    href="/order/order-1"
    onClick={[Function]}
    onMouseOver={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    title={null}
  >
    <div
      className="rowNotificationDot"
    />
    <div
      className="itemUsername"
    >
      <span
        className=""
      >
        provider-user-id display name
      </span>
    </div>
    <div
      className="itemTitle"
    >
      ItemX title
    </div>
    <div
      className="itemDetails"
    />
    <div
      className="itemState"
    >
      <div
        className="stateName stateNoActionNeeded"
      >
        <span>
          InboxPage.default-buying-products.purchased.status
        </span>
      </div>
    </div>
  </a>
</div>
`;

exports[`InboxPage InboxItem matches snapshot of sales 1`] = `
<div
  className="item"
>
  <div
    className="itemAvatar"
  >
    <a
      className="root"
      href="/u/customer-user-id"
      onClick={[Function]}
      onMouseOver={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
      title="customer-user-id display name"
    >
      <span
        className="initials"
      >
        TT
      </span>
    </a>
  </div>
  <a
    className="itemLink"
    href="/sale/sale-1"
    onClick={[Function]}
    onMouseOver={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    title={null}
  >
    <div
      className="rowNotificationDot"
    >
      <div
        className="notificationDot"
      />
    </div>
    <div
      className="itemUsername"
    >
      <span
        className=""
      >
        customer-user-id display name
      </span>
    </div>
    <div
      className="itemTitle"
    >
      ItemX title
    </div>
    <div
      className="itemDetails"
    />
    <div
      className="itemState"
    >
      <div
        className="stateName stateActionNeeded"
      >
        <span>
          InboxPage.default-buying-products.purchased.status
        </span>
      </div>
    </div>
  </a>
</div>
`;

exports[`InboxPage InboxPageComponent matches snapshot of order 1`] = `
<Page
  scrollingDisabled={false}
  title="InboxPage.ordersTitle"
>
  <LayoutSideNavigation
    className={null}
    containerClassName={null}
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent))
        className="topbar"
        currentPage="InboxPage"
        desktopClassName="desktopTopbar"
        mobileRootClassName="mobileTopbar"
      />
    </LayoutWrapperTopbar>
    <LayoutWrapperSideNav
      className="navigation"
      rootClassName={null}
      tabs={null}
    >
      <h1
        className="title"
      >
        <MemoizedFormattedMessage
          id="InboxPage.title"
        />
      </h1>
      <TabNav
        className={null}
        rootClassName="tabs"
        tabClassName={null}
        tabRootClassName="tab"
        tabs={
          Array [
            Object {
              "linkProps": Object {
                "name": "InboxPage",
                "params": Object {
                  "tab": "orders",
                },
              },
              "selected": true,
              "text": <span>
                <Memo(MemoizedFormattedMessage)
                  id="InboxPage.ordersTabTitle"
                />
              </span>,
            },
            Object {
              "linkProps": Object {
                "name": "InboxPage",
                "params": Object {
                  "tab": "sales",
                },
              },
              "selected": false,
              "text": <span>
                <Memo(MemoizedFormattedMessage)
                  id="InboxPage.salesTabTitle"
                />
              </span>,
            },
          ]
        }
      />
    </LayoutWrapperSideNav>
    <LayoutWrapperMain
      className={null}
      rootClassName={null}
    >
      <ul
        className="itemList"
      >
        <li
          className="listItem"
          key="order-1"
        >
          <InboxItem
            intl={
              Object {
                "formatDate": [Function],
                "formatDateTimeRange": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            isBooking={false}
            stateData={
              Object {
                "processName": "default-buying-products",
                "processState": "purchased",
              }
            }
            transactionRole="customer"
            tx={
              Object {
                "attributes": Object {
                  "createdAt": 2017-05-01T00:00:00.000Z,
                  "lastTransition": "transition/confirm-payment",
                  "lastTransitionedAt": 2017-01-15T00:00:00.000Z,
                  "lineItems": Array [
                    Object {
                      "code": "line-item/item",
                      "includeFor": Array [
                        "customer",
                        "provider",
                      ],
                      "lineTotal": Money {
                        "_sdkType": "Money",
                        "amount": 1000,
                        "currency": "USD",
                      },
                      "quantity": "1",
                      "reversal": false,
                      "unitPrice": Money {
                        "_sdkType": "Money",
                        "amount": 1000,
                        "currency": "USD",
                      },
                    },
                    Object {
                      "code": "line-item/provider-commission",
                      "includeFor": Array [
                        "provider",
                      ],
                      "lineTotal": Money {
                        "_sdkType": "Money",
                        "amount": -100,
                        "currency": "USD",
                      },
                      "reversal": false,
                      "unitPrice": Money {
                        "_sdkType": "Money",
                        "amount": -100,
                        "currency": "USD",
                      },
                    },
                  ],
                  "payinTotal": Money {
                    "_sdkType": "Money",
                    "amount": 1000,
                    "currency": "USD",
                  },
                  "payoutTotal": Money {
                    "_sdkType": "Money",
                    "amount": 900,
                    "currency": "USD",
                  },
                  "processName": "default-buying-products",
                  "processVersion": 1,
                  "transitions": Array [
                    Object {
                      "by": "customer",
                      "createdAt": 2017-05-01T00:00:00.000Z,
                      "transition": "transition/request-payment",
                    },
                    Object {
                      "by": "customer",
                      "createdAt": 2017-05-01T00:00:01.000Z,
                      "transition": "transition/confirm-payment",
                    },
                  ],
                },
                "booking": null,
                "customer": Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "profile": Object {
                      "abbreviatedName": "TT",
                      "displayName": "customer-user-id display name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "customer-user-id",
                  },
                  "type": "user",
                },
                "id": UUID {
                  "_sdkType": "UUID",
                  "uuid": "order-1",
                },
                "listing": Object {
                  "attributes": Object {
                    "deleted": false,
                    "description": "ItemX description",
                    "geolocation": LatLng {
                      "_sdkType": "LatLng",
                      "lat": 40,
                      "lng": 60,
                    },
                    "price": Money {
                      "_sdkType": "Money",
                      "amount": 5500,
                      "currency": "USD",
                    },
                    "publicData": Object {
                      "transactionProcessAlias": "default-buying-products",
                      "transactionType": "sell-bikes",
                      "unitType": "item",
                    },
                    "state": "published",
                    "title": "ItemX title",
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "ItemX",
                  },
                  "type": "listing",
                },
                "provider": Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "profile": Object {
                      "abbreviatedName": "TT",
                      "displayName": "provider-user-id display name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "provider-user-id",
                  },
                  "type": "user",
                },
                "reviews": Array [],
                "type": "transaction",
              }
            }
          />
        </li>
        <li
          className="listItem"
          key="order-2"
        >
          <InboxItem
            intl={
              Object {
                "formatDate": [Function],
                "formatDateTimeRange": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            isBooking={false}
            stateData={
              Object {
                "processName": "default-buying-products",
                "processState": "purchased",
              }
            }
            transactionRole="customer"
            tx={
              Object {
                "attributes": Object {
                  "createdAt": 2017-05-01T00:00:00.000Z,
                  "lastTransition": "transition/confirm-payment",
                  "lastTransitionedAt": 2016-01-15T00:00:00.000Z,
                  "lineItems": Array [
                    Object {
                      "code": "line-item/item",
                      "includeFor": Array [
                        "customer",
                        "provider",
                      ],
                      "lineTotal": Money {
                        "_sdkType": "Money",
                        "amount": 1000,
                        "currency": "USD",
                      },
                      "quantity": "1",
                      "reversal": false,
                      "unitPrice": Money {
                        "_sdkType": "Money",
                        "amount": 1000,
                        "currency": "USD",
                      },
                    },
                    Object {
                      "code": "line-item/provider-commission",
                      "includeFor": Array [
                        "provider",
                      ],
                      "lineTotal": Money {
                        "_sdkType": "Money",
                        "amount": -100,
                        "currency": "USD",
                      },
                      "reversal": false,
                      "unitPrice": Money {
                        "_sdkType": "Money",
                        "amount": -100,
                        "currency": "USD",
                      },
                    },
                  ],
                  "payinTotal": Money {
                    "_sdkType": "Money",
                    "amount": 1000,
                    "currency": "USD",
                  },
                  "payoutTotal": Money {
                    "_sdkType": "Money",
                    "amount": 900,
                    "currency": "USD",
                  },
                  "processName": "default-buying-products",
                  "processVersion": 1,
                  "transitions": Array [
                    Object {
                      "by": "customer",
                      "createdAt": 2017-05-01T00:00:00.000Z,
                      "transition": "transition/request-payment",
                    },
                    Object {
                      "by": "customer",
                      "createdAt": 2017-05-01T00:00:01.000Z,
                      "transition": "transition/confirm-payment",
                    },
                  ],
                },
                "booking": null,
                "customer": Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "profile": Object {
                      "abbreviatedName": "TT",
                      "displayName": "customer-user-id display name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "customer-user-id",
                  },
                  "type": "user",
                },
                "id": UUID {
                  "_sdkType": "UUID",
                  "uuid": "order-2",
                },
                "listing": Object {
                  "attributes": Object {
                    "deleted": false,
                    "description": "ItemX description",
                    "geolocation": LatLng {
                      "_sdkType": "LatLng",
                      "lat": 40,
                      "lng": 60,
                    },
                    "price": Money {
                      "_sdkType": "Money",
                      "amount": 5500,
                      "currency": "USD",
                    },
                    "publicData": Object {
                      "transactionProcessAlias": "default-buying-products",
                      "transactionType": "sell-bikes",
                      "unitType": "item",
                    },
                    "state": "published",
                    "title": "ItemX title",
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "ItemX",
                  },
                  "type": "listing",
                },
                "provider": Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "profile": Object {
                      "abbreviatedName": "TT",
                      "displayName": "provider-user-id display name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "provider-user-id",
                  },
                  "type": "user",
                },
                "reviews": Array [],
                "type": "transaction",
              }
            }
          />
        </li>
      </ul>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSideNavigation>
</Page>
`;

exports[`InboxPage InboxPageComponent matches snapshot of sales 1`] = `
<Page
  scrollingDisabled={false}
  title="InboxPage.salesTitle"
>
  <LayoutSideNavigation
    className={null}
    containerClassName={null}
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent))
        className="topbar"
        currentPage="InboxPage"
        desktopClassName="desktopTopbar"
        mobileRootClassName="mobileTopbar"
      />
    </LayoutWrapperTopbar>
    <LayoutWrapperSideNav
      className="navigation"
      rootClassName={null}
      tabs={null}
    >
      <h1
        className="title"
      >
        <MemoizedFormattedMessage
          id="InboxPage.title"
        />
      </h1>
      <TabNav
        className={null}
        rootClassName="tabs"
        tabClassName={null}
        tabRootClassName="tab"
        tabs={
          Array [
            Object {
              "linkProps": Object {
                "name": "InboxPage",
                "params": Object {
                  "tab": "orders",
                },
              },
              "selected": false,
              "text": <span>
                <Memo(MemoizedFormattedMessage)
                  id="InboxPage.ordersTabTitle"
                />
              </span>,
            },
            Object {
              "linkProps": Object {
                "name": "InboxPage",
                "params": Object {
                  "tab": "sales",
                },
              },
              "selected": true,
              "text": <span>
                <Memo(MemoizedFormattedMessage)
                  id="InboxPage.salesTabTitle"
                />
              </span>,
            },
          ]
        }
      />
    </LayoutWrapperSideNav>
    <LayoutWrapperMain
      className={null}
      rootClassName={null}
    >
      <ul
        className="itemList"
      >
        <li
          className="listItem"
          key="sale-1"
        >
          <InboxItem
            intl={
              Object {
                "formatDate": [Function],
                "formatDateTimeRange": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            isBooking={false}
            stateData={
              Object {
                "actionNeeded": true,
                "isSaleNotification": true,
                "processName": "default-buying-products",
                "processState": "purchased",
              }
            }
            transactionRole="provider"
            tx={
              Object {
                "attributes": Object {
                  "createdAt": 2017-05-01T00:00:00.000Z,
                  "lastTransition": "transition/confirm-payment",
                  "lastTransitionedAt": 2017-01-15T00:00:00.000Z,
                  "lineItems": Array [
                    Object {
                      "code": "line-item/item",
                      "includeFor": Array [
                        "customer",
                        "provider",
                      ],
                      "lineTotal": Money {
                        "_sdkType": "Money",
                        "amount": 1000,
                        "currency": "USD",
                      },
                      "quantity": "1",
                      "reversal": false,
                      "unitPrice": Money {
                        "_sdkType": "Money",
                        "amount": 1000,
                        "currency": "USD",
                      },
                    },
                    Object {
                      "code": "line-item/provider-commission",
                      "includeFor": Array [
                        "provider",
                      ],
                      "lineTotal": Money {
                        "_sdkType": "Money",
                        "amount": -100,
                        "currency": "USD",
                      },
                      "reversal": false,
                      "unitPrice": Money {
                        "_sdkType": "Money",
                        "amount": -100,
                        "currency": "USD",
                      },
                    },
                  ],
                  "payinTotal": Money {
                    "_sdkType": "Money",
                    "amount": 1000,
                    "currency": "USD",
                  },
                  "payoutTotal": Money {
                    "_sdkType": "Money",
                    "amount": 900,
                    "currency": "USD",
                  },
                  "processName": "default-buying-products",
                  "processVersion": 1,
                  "transitions": Array [
                    Object {
                      "by": "customer",
                      "createdAt": 2017-05-01T00:00:00.000Z,
                      "transition": "transition/request-payment",
                    },
                    Object {
                      "by": "customer",
                      "createdAt": 2017-05-01T00:00:01.000Z,
                      "transition": "transition/confirm-payment",
                    },
                  ],
                },
                "booking": null,
                "customer": Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "profile": Object {
                      "abbreviatedName": "TT",
                      "displayName": "customer-user-id display name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "customer-user-id",
                  },
                  "type": "user",
                },
                "id": UUID {
                  "_sdkType": "UUID",
                  "uuid": "sale-1",
                },
                "listing": Object {
                  "attributes": Object {
                    "deleted": false,
                    "description": "ItemX description",
                    "geolocation": LatLng {
                      "_sdkType": "LatLng",
                      "lat": 40,
                      "lng": 60,
                    },
                    "price": Money {
                      "_sdkType": "Money",
                      "amount": 5500,
                      "currency": "USD",
                    },
                    "publicData": Object {
                      "transactionProcessAlias": "default-buying-products",
                      "transactionType": "sell-bikes",
                      "unitType": "item",
                    },
                    "state": "published",
                    "title": "ItemX title",
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "ItemX",
                  },
                  "type": "listing",
                },
                "provider": Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "profile": Object {
                      "abbreviatedName": "TT",
                      "displayName": "provider-user-id display name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "provider-user-id",
                  },
                  "type": "user",
                },
                "reviews": Array [],
                "type": "transaction",
              }
            }
          />
        </li>
        <li
          className="listItem"
          key="sale-2"
        >
          <InboxItem
            intl={
              Object {
                "formatDate": [Function],
                "formatDateTimeRange": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            isBooking={false}
            stateData={
              Object {
                "actionNeeded": true,
                "isSaleNotification": true,
                "processName": "default-buying-products",
                "processState": "purchased",
              }
            }
            transactionRole="provider"
            tx={
              Object {
                "attributes": Object {
                  "createdAt": 2017-05-01T00:00:00.000Z,
                  "lastTransition": "transition/confirm-payment",
                  "lastTransitionedAt": 2016-01-15T00:00:00.000Z,
                  "lineItems": Array [
                    Object {
                      "code": "line-item/item",
                      "includeFor": Array [
                        "customer",
                        "provider",
                      ],
                      "lineTotal": Money {
                        "_sdkType": "Money",
                        "amount": 1000,
                        "currency": "USD",
                      },
                      "quantity": "1",
                      "reversal": false,
                      "unitPrice": Money {
                        "_sdkType": "Money",
                        "amount": 1000,
                        "currency": "USD",
                      },
                    },
                    Object {
                      "code": "line-item/provider-commission",
                      "includeFor": Array [
                        "provider",
                      ],
                      "lineTotal": Money {
                        "_sdkType": "Money",
                        "amount": -100,
                        "currency": "USD",
                      },
                      "reversal": false,
                      "unitPrice": Money {
                        "_sdkType": "Money",
                        "amount": -100,
                        "currency": "USD",
                      },
                    },
                  ],
                  "payinTotal": Money {
                    "_sdkType": "Money",
                    "amount": 1000,
                    "currency": "USD",
                  },
                  "payoutTotal": Money {
                    "_sdkType": "Money",
                    "amount": 900,
                    "currency": "USD",
                  },
                  "processName": "default-buying-products",
                  "processVersion": 1,
                  "transitions": Array [
                    Object {
                      "by": "customer",
                      "createdAt": 2017-05-01T00:00:00.000Z,
                      "transition": "transition/request-payment",
                    },
                    Object {
                      "by": "customer",
                      "createdAt": 2017-05-01T00:00:01.000Z,
                      "transition": "transition/confirm-payment",
                    },
                  ],
                },
                "booking": null,
                "customer": Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "profile": Object {
                      "abbreviatedName": "TT",
                      "displayName": "customer-user-id display name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "customer-user-id",
                  },
                  "type": "user",
                },
                "id": UUID {
                  "_sdkType": "UUID",
                  "uuid": "sale-2",
                },
                "listing": Object {
                  "attributes": Object {
                    "deleted": false,
                    "description": "ItemX description",
                    "geolocation": LatLng {
                      "_sdkType": "LatLng",
                      "lat": 40,
                      "lng": 60,
                    },
                    "price": Money {
                      "_sdkType": "Money",
                      "amount": 5500,
                      "currency": "USD",
                    },
                    "publicData": Object {
                      "transactionProcessAlias": "default-buying-products",
                      "transactionType": "sell-bikes",
                      "unitType": "item",
                    },
                    "state": "published",
                    "title": "ItemX title",
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "ItemX",
                  },
                  "type": "listing",
                },
                "provider": Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "profile": Object {
                      "abbreviatedName": "TT",
                      "displayName": "provider-user-id display name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "provider-user-id",
                  },
                  "type": "user",
                },
                "reviews": Array [],
                "type": "transaction",
              }
            }
          />
        </li>
      </ul>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSideNavigation>
</Page>
`;
